### PR TITLE
Fix the "compare" Command

### PR DESCRIPTION
**Pull request by ericsnowcurrently**
Looks like some bugs crept in with the mypy changes.

### CODE REVIEW

File: PORTAL/jobs/__main__.py [hunk 1]
```diff
@@ -1076,6 +1076,16 @@ def add_cmd(name, subs=subs, *, parents=(), **kwargs):
     return cmd, ns, cfgfile, user, verbosity, logfile, devmode
 
 
+def _should_ensure_next(cmd):
+    # In some cases the mechanism to run jobs from the queue may
+    # get interrupted, so we re-start it manually here if necessary.
+    if cmd in ('queue-info', 'compare'):
+        return False
```
Comment (ericsnowcurrently) [inline PORTAL/jobs/__main__.py:1083]: Something is broken for `ensure_next()`.

```
$ bench-jobs-dev compare req-compile-bench-1663116583-esnow req-compile-bench-1663116597-esnow
Traceback (most recent call last):
  File "/usr/lib/python3.8/runpy.py", line 192, in _run_module_as_main
    return _run_code(code, main_globals, None,
  File "/usr/lib/python3.8/runpy.py", line 85, in _run_code
    exec(code, run_globals)
  File "/home/esnow/faster-cpython-tools/PORTAL/jobs/__main__.py", line 1177, in <module>
    main(cmd, cmd_kwargs, cfgfile, user, devmode)
  File "/home/esnow/faster-cpython-tools/PORTAL/jobs/__main__.py", line 1126, in main
    jobs.ensure_next()
  File "/home/esnow/faster-cpython-tools/PORTAL/jobs/__init__.py", line 272, in ensure_next
    queue = self.queue.snapshot
  File "/home/esnow/faster-cpython-tools/PORTAL/jobs/queue.py", line 246, in snapshot
    assert pid is not None
AssertionError
```

I'll open a separate issue for this.  In the meantime this workaround should unblock me.

**Review by mdboom — APPROVED** [review]
LGTM.

It would be good to write a test for this, but I think it would be tricky without a way to mock up the benchmarking server.  Something for the future, for sure.

**Comment by gvanrossum** [comment]
Would it be prudent to add a test that would have discovered the breakage?

**Comment by mdboom** [comment]
> Would it be prudent to add a test that would have discovered the breakage?

Yeah, I totally agree.  In this case of these high-level methods, we'll need something to mock-up the responses from the benchmarking server, which is a bit out-of-scope for getting this fix merged.  I filed #104 to get that going (and there is a broader effort to add tests started in #97).

**Pull request merged**